pkg/vault/instances.go

```diff
@@ -188,14 +202,26 @@ func configureMaster(instanceCreds map[string]AuthBundle) string {
 			roleID := mustGetenv("VAULT_ROLE_ID")
 			secretID := mustGetenv("VAULT_SECRET_ID")
 
-			secret, err := client.Logical().Write("auth/approle/login", map[string]interface{}{
-				"role_id":   roleID,
-				"secret_id": secretID,
+			err := utils.Retry(defaultTokenRetryAttempts, defaultTokenRetrySleep, func() error {
+				secret, err := client.Logical().Write("auth/approle/login", map[string]interface{}{
```

Comment:
The hashicorp vault client has login methods, that take care of this: https://developer.hashicorp.com/vault/docs/auth/approle#code-example

Reply:
@janboll, the API is much nicer (such as many convenience wrappers and added validation, etc.).

I will switch to it! Thank you!

Reply:
@hemslo had a chat with me about this, where he voiced some reservations about adding a retry, especially since this behaviour we see from Vault is of a grey area: it's not a documented error (also transient, as subsequent request often succeeds), and there aren't any guidelines from HashiCorp about what users should do if the authentication information is missing (why this is happening hasn't been explained in the documentation) - except for the newer API (which is a more high level than the one we are using at the moment), the one that @janboll is using and suggested we switch to, where lack of client token is then returned as an error.